src/sidebar/components/test/group-list-test.js

```diff
@@ -363,6 +385,125 @@ describe('groupList', function() {
       '_blank'
     );
   });
+  describe('group section visibility', () => {
+    [
+      {
+        description:
+          'shows Currently Viewing section when there are currently viewing groups',
+        currentlyViewingGroups: [publicGroup],
+        featuredGroups: [restrictedGroup],
+        myGroups: [],
+        expectedSections: ['Currently Viewing', 'Featured Groups'],
+      },
+      {
+        description:
+          'shows Featured Groups section when there are featured groups',
+        currentlyViewingGroups: [],
+        featuredGroups: [restrictedGroup],
+        myGroups: [publicGroup],
+        expectedSections: ['Featured Groups', 'My Groups'],
+      },
+      {
+        description: 'shows My Groups section when there are my groups',
+        currentlyViewingGroups: [],
+        featuredGroups: [],
+        myGroups: [publicGroup, privateGroup],
+        expectedSections: ['My Groups'],
+      },
+    ].forEach(
+      ({
+        description,
+        currentlyViewingGroups,
+        featuredGroups,
+        myGroups,
+        expectedSections,
+      }) => {
+        it(description, () => {
+          fakeFeatures.flagEnabled.withArgs('community_groups').returns(true);
+          // In order to show the group drop down there must be at least two groups.
+          groups = currentlyViewingGroups
+            .concat(featuredGroups)
+            .concat(myGroups);
+          fakeStore.getCurrentlyViewingGroups.returns(currentlyViewingGroups);
+          fakeStore.getFeaturedGroups.returns(featuredGroups);
+          fakeStore.getMyGroups.returns(myGroups);
+
+          const element = createGroupList();
+
+          const showGroupsMenu = element.ctrl.showGroupsMenu();
+          const dropdownToggle = element.find('.dropdown-toggle');
+          const arrowIcon = element.find('.h-icon-arrow-drop-down');
+          const sectionHeader = element.find('.dropdown-menu__section-heading');
+          const section = element.find('.dropdown-menu__section');
+          const dropdownOptions = element.find(
+            '.dropdown-community-groups-menu__row'
+          );
+
+          assert.isTrue(showGroupsMenu);
+          assert.lengthOf(dropdownToggle, 1);
+          assert.lengthOf(arrowIcon, 1);
+          sectionHeader.each(function() {
+            assert.isTrue(expectedSections.includes(this.textContent));
+          });
+          // Plus one for the create private group section.
+          assert.lengthOf(section, expectedSections.length + 1);
+          assert.lengthOf(dropdownOptions, 3);
+        });
+      }
+    );
+  });
+  describe('group details expanded on out of scope groups', () => {
+    it('sets the default for the given groupid to false and returns it', () => {
+      const element = createGroupList();
+
+      const expanded = element.ctrl.isGroupDetailsExpanded('groupid');
+
+      assert.isFalse(expanded);
+      assert.isFalse(element.ctrl.groupDetailsExpanded.groupid);
```

Comment:
I would avoid testing internal properties in tests, as it creates more work if you later change the implementation but not the user-facing behaviour.